src/ext/BalExtension/wixstdba/WixStandardBootstrapperApplication.cpp

```diff
@@ -934,6 +952,26 @@ class CWixStandardBootstrapperApplication : public CBalBaseBootstrapperApplicati
         }
     }
 
+    virtual STDMETHODIMP_(int) OnExecuteFilesInUse(
+        __in_z LPCWSTR wzPackageId,
+        __in DWORD cFiles,
+        __in_ecount_z(cFiles) LPCWSTR* rgwzFiles
+        )
+    {
+        if (m_fShowFilesInUse && !m_fPrereq && wzPackageId && *wzPackageId)
+        {
+            //If this is an MSI package, display the files in use page.
+            BAL_INFO_PACKAGE* pPackage = NULL;
+            BalInfoFindPackageById(&m_Bundle.packages, wzPackageId, &pPackage);
+
+            if (pPackage && BAL_INFO_PACKAGE_TYPE_MSI == pPackage->type)
+            {
+                return ShowFilesInUseModal(cFiles, rgwzFiles);
```

Comment:
Calling it modal is really confusing to me when it's just another page.  I'd really like to see the new page be a modal window instead.